src/pudl/extract/excel.py

```diff
@@ -287,8 +287,8 @@ def extract(self, **partitions):
 
             # After all years are loaded, add empty columns that could appear
             # in other years so that df matches the database schema
-            missing_cols = set(self._metadata.get_all_columns(page)).difference(
-                df.columns
+            missing_cols = sorted(
```

Comment:
Hmm okay, I didn't realize sets always turned into sorted lists. I've been trying to use `sorted()` whenever it's going to be in an error message or otherwise visible so it's easy find elements, and figured no ordering was inherent in a `set` since it's not an ordered collection.